### DISCUSSION TITLE
GSOC

### DISCUSSION THREAD

**Discussion by Ali7040** [stadiamaps/ferrostar#57] [Engineering RFDs]
hi there!

I reached out because your project was mentioned on wiki.openstreetmap.org as selected for Google Summer of Code. The project requirements include JavaScript, frontend frameworks (React preferred for maximum reach), and some familiarity with binding generation. Please consider me for any frontend-related issues or issues that require JavaScript or TypeScript expertise.

**Comment by ianthetechie** [comment]
Thanks @Ali7040!

To give a bit more context for the GSoC project, I'd love to be able to offer support for foreground navigation in PWAs (progressive web apps). Have a read over #49 and let me know if you have any thoughts on how best to approach this, as well as any limits that I haven't thought of yet which would need consideration to create a good PWA experience.

Finally, I'd also be interested in your opinions on the best way to distribute the resulting frontend as a library (I guess a React component of some sort?) and what integration would look like for end users.
